source/slang/slang-parser.cpp

```diff
@@ -4829,25 +4850,54 @@ namespace Slang
             // a conjunction type expression.
 
             auto loc = peekToken(parser).loc;
-            if(!AdvanceIf(parser, TokenType::OpBitAnd))
-                break;
+            if(AdvanceIf(parser, TokenType::OpBitAnd))
+            {
+                auto rightExpr = _parsePostfixTypeExpr(parser);
+
+                auto andExpr = parser->astBuilder->create<AndTypeExpr>();
+                andExpr->loc = loc;
+                andExpr->left = TypeExp(leftExpr);
+                andExpr->right = TypeExp(rightExpr);
+                leftExpr = andExpr;
+            }
```

Comment:
how about `func (a,b) -> c`. This mirrors nicely the syntax we have already for function declarations.

Reply:
This may cause issues if we decide to use the func keyword to define lambdas. I think Func<R, T> is safest for now and we can decide a more convenient syntax later.